integration/src/test/java/edu/unc/lib/boxc/integration/web/access/SearchEndpointIT.java

```diff
@@ -348,16 +347,58 @@ public void testSearchWithCreatedDateRangeSpecifiedAsUnknown() throws Exception
         }
     }
 
-    private List<String> createDatedObjects() throws Exception {
-        List<String> ids = new ArrayList<>();
-        var datedAdminUnit = adminUnitFactory.createAdminUnit(
-                Map.of("title", "Dated Admin Object", "dateCreated", "2018-07-01"));
-        var datedCollection = collectionFactory.createCollection(datedAdminUnit,
-                Map.of("title", "A dated collection",
-                        "dateCreated", "2022-07-01",
-                        "readGroup", "everyone"));
-        ids.add(datedAdminUnit.getPid().getId());
-        ids.add(datedCollection.getPid().getId());
-        return ids;
+    @Test
+    public void testSearchWithOneFacetFilter() throws Exception {
+        createDefaultObjects();
+        createLanguageSubjectObjects();
+
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language&language=English&getFacets=true");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var facetFields = getNodeFromResponse(resp, "facetFields");
+
+            assertSuccessfulResponse(resp);
+            // should only find the language facet
+            assertEquals(1, facetFields.size());
+        }
+    }
+
+    @Test
+    public void testSearchWithLanguageFilterSelectedDisplaysAllAvailableLanguages() throws Exception {
+        createDefaultObjects();
+        createLanguageSubjectObjects();
+
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language&language=English&getFacets=true");
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var facetFields = getNodeFromResponse(resp, "facetFields");
+            var languageFields = new ArrayList<JsonNode>();
+            facetFields.get(0).get("values").elements().forEachRemaining(languageFields::add);
+
+            assertSuccessfulResponse(resp);
+            // should find all available languages: English, Cherokee
+            assertValuePresent(languageFields,0,"value", "English");
+            assertValuePresent(languageFields,1,"value", "Cherokee");
+        }
+    }
+
+    @Test
+    public void testSearchWithSubjectFilterSelectedDisplaysAvailableLanguages() throws Exception {
+        createDefaultObjects();
+        createLanguageSubjectObjects();
+
+        var getMethod = new HttpGet(SEARCH_URL + "/?facetSelect=language%2Csubject&subject=North%2520Carolina&getFacets=true");
```

Comment:
I'm not sure why the access app is double encoding the spaces, looks like it can be either `%20` or `+`, but that is indeed what it is doing, hmm. Maybe we can correct that in the future.

Reply:
I have no idea what this means

Reply:
Just remarking on `North%2520Carolina`, which contains a space that normally would be encoded as a `+` or `%20`, but has been double encoded from `%2520` (the % got encoded to `%25`). Its not a problem with this PR, just a not great thing

Reply:
ah, is that just something @krwong added manually? or does it actually look like this on prod?

Reply:
It looks like that on prod, but it works with any of the 3 encodings i listed.